bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableTableAdminClientWrapper.java

```diff
@@ -306,4 +317,108 @@ public Void apply(Empty input) {
+    GetBackupRequest request =
+        GetBackupRequest.newBuilder().setName(clusterName.toBackupName(backupId)).build();
+    return ApiFutureUtil.transformAndAdapt(
+        delegate.getBackupAsync(request),
+        new Function<com.google.bigtable.admin.v2.Backup, Backup>() {
+          @Override
+          public Backup apply(com.google.bigtable.admin.v2.Backup backup) {
+            return Backup.fromProto(backup);
+          }
+        });
+  }
+
+  @Override
+  public ApiFuture<Backup> updateBackupAsync(UpdateBackupRequest request) {
+    return ApiFutureUtil.transformAndAdapt(
+        delegate.updateBackupAsync(
+            request.toProto(instanceName.getProjectId(), instanceName.getInstanceId())),
+        new Function<com.google.bigtable.admin.v2.Backup, Backup>() {
+          @Override
+          public Backup apply(com.google.bigtable.admin.v2.Backup backup) {
+            return Backup.fromProto(backup);
+          }
+        });
+  }
+
+  @Override
+  public ApiFuture<List<String>> listBackupsAsync(String clusterId) {
+    BigtableClusterName clusterName = instanceName.toClusterName(clusterId);
+    ListBackupsRequest request =
+        ListBackupsRequest.newBuilder().setParent(clusterName.getClusterName()).build();
+
+    return ApiFutureUtil.transformAndAdapt(
+        delegate.listBackupsAsync(request),
+        new Function<ListBackupsResponse, List<String>>() {
+          @Override
+          public List<String> apply(ListBackupsResponse response) {
+            List<String> backups = new ArrayList<>();
+            for (com.google.bigtable.admin.v2.Backup backup : response.getBackupsList()) {
+              backups.add(backup.getName());
+            }
+            return backups;
+          }
+        });
+  }
+
+  @Override
+  public ApiFuture<Void> deleteBackupAsync(String clusterId, String backupId) {
+    BigtableClusterName clusterName = instanceName.toClusterName(clusterId);
+    DeleteBackupRequest request =
+        DeleteBackupRequest.newBuilder().setName(clusterName.toBackupName(backupId)).build();
+    return ApiFutureUtil.transformAndAdapt(
+        delegate.deleteBackupAsync(request),
+        new Function<Empty, Void>() {
+          @Override
+          public Void apply(Empty empty) {
+            return null;
+          }
+        });
+  }
+
+  @Override
+  public ApiFuture<RestoredTableResult> restoreTableAsync(RestoreTableRequest request) {
+    return ApiFutureUtil.transformAndAdapt(
+        delegate.restoreTableAsync(
+            request.toProto(instanceName.getProjectId(), instanceName.getInstanceId())),
+        new Function<Operation, RestoredTableResult>() {
+          @Override
+          public RestoredTableResult apply(Operation operation) {
+            try {
+              return new RestoredTableResult(
+                  Table.fromProto(
+                      operation.getResponse().unpack(com.google.bigtable.admin.v2.Table.class)),
```

Comment:
Same problem as above, the operation is not yet complete....you need to wait on it for it to have a payload/response